File: jmlparser-jml-tools/src/main/java/com/github/jmlparser/xpath/PseudoAttributeProvider.java
package com.github.jmlparser.xpath;

import com.github.javaparser.ast.Node;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.w3c.dom.Element;

import java.util.Collection;

/**
 * @author Alexander Weigl
 * @version 1 (11.02.23)
 */
public interface PseudoAttributeProvider {
    @Nullable
    Collection<JPAttrPseudo> attributeForNode(@NotNull Node node, @NotNull Element owner);
}

File: tools/xpath/src/main/kotlin/io/github/jmltoolkit/xpath/PseudoAttributeProvider.kt
package io.github.jmltoolkit.xpath

import com.github.javaparser.ast.Node
import org.w3c.dom.Element

/**
 * @author Alexander Weigl
 * @version 1 (11.02.23)
 */
interface PseudoAttributeProvider {
    fun attributeForNode(node: Node, owner: Element): Collection<JPAttrPseudo>
}
